src/main.rs

```diff
@@ -23,30 +28,84 @@ macro_rules! run_action {
     }};
 }
 
+fn read_os_release<P: AsRef<Path>>(zroot: P) -> io::Result<OsRelease> {
+    let zroot = zroot.as_ref();
+    let path = &zroot.join("etc/os-release");
+    let release = OsRelease::new_from(path)?;
+    Ok(release)
+}
+
+fn get_zfs_parent(s: &str) -> String {
+    let zonename = zonename::getzonename().expect("failed to get zonename");
+    if s == "" {
+        if zonename == "global" {
+            return "zones".to_string();
+        } else {
+            return format!("zones/{}/data", &zonename);
+        }
+    } else {
+        return s.to_string();
+    }
+}
+
 fn main() -> Result<()> {
     let opts = cli::get_opts();
-
     let utc: DateTime<Utc> = Utc::now();
     let build_date = utc.format("%Y%m%d").to_string();
-    let iuuid = format!("{}-{}", &opts.name, &build_date);
-    let dataset = format!("{}/{}", &opts.zfs_parent, &iuuid);
-    let zfs_tar = format!("{}.zfs.gz", &iuuid);
-    let image_manifest = &format!("{}.json", &iuuid);
+    let uuid = Uuid::new_v4();
+    let iuuid = format!("{}-{}", &uuid, &build_date);
+    let zfs_parent = get_zfs_parent(&opts.zfs_parent);
+
+    let dataset = format!("{}/{}", &zfs_parent, &iuuid);
+    let zroot = create_dataset(&dataset)?;
+    run_action!(install_tar(&zroot, &opts.tar), &dataset);
+
+    let os_release = read_os_release(&zroot).unwrap();
+    let name = format!("{}-{}", os_release.id, os_release.version_id)
+        .trim_end_matches("-")
+        .to_string();
+    let zfs_tar = format!("{}-{}.zfs.gz", &name, &build_date);
+    let image_manifest = &format!("{}-{}.json", &name, &build_date);
+
+    let desc = format!(
+        "Container-native {} 64-bit image. {}",
+        os_release.pretty_name, &opts.description
+    );
     let manifest = Manifest {
-        name: &opts.name,
+        name: &name,
         version: &build_date,
-        description: &opts.description,
+        description: &desc.trim(),
         homepage: &opts.url,
         min_platform: &opts.min_platform,
-        uuid: &Uuid::new_v4(),
+        uuid: &uuid,
         os: "linux",
         kernel: &opts.kernel,
         tar_file: &zfs_tar,
     };
+    let product = [
```

Comment:
You could use literal strings along with [captured identifiers](https://blog.rust-lang.org/2022/01/13/Rust-1.58.0.html#captured-identifiers-in-format-strings) for product and motd.  Example:
```rust
fn main() {
    let pretty_name = "foo";
    let build_date = "bar";
    let url = "baz";
    let desc = "example";

    let product = format!(
        r#"Name: Joyent Instance
Image: {pretty_name} {build_date}
Documentation: {url}
Description: {desc}

"#
    );

    println!("{}", product);
}```

Produces:

```❯ cargo r
   Compiling scratch v0.1.0 (/private/tmp/scratch)
    Finished dev [unoptimized + debuginfo] target(s) in 0.33s
     Running `target/debug/scratch`
Name: Joyent Instance
Image: foo bar
Documentation: baz
Description: example

```

Reply:
Fixed.